appveyor.yml

```diff
@@ -1,11 +1,65 @@
 image: Visual Studio 2015
 
+environment:
+  NUGET_API_KEY:
+    secure: Rh/UHYn7HGj8a7LFWo9C1rSA0sHSQNV3wIqj9bvwRR8yLmCXu9JmbNkmkctk2cBn
+  MYGET_API_KEY:
+    secure: hA4Ut1N2lrrdEtAN24Bty/FNiU0d/Ur/dLYSqpr8jKHOvoO7MU4jD+KwzUvATh+E
+
 pull_requests:
   do_not_increment_build_number: true
 
 build_script:
 - ps: |
-    & .\build.cmd CompleteBuild
+    # A trick to test if variable is defined. Number could never be null.
+    $isPullRequest = -not ($Env:APPVEYOR_PULL_REQUEST_NUMBER -eq $null)
+    $isTagBuild = $Env:APPVEYOR_REPO_TAG -eq "true"
+    $branchName = $Env:APPVEYOR_REPO_BRANCH
+
+    Write-Host "[BUILD CONTEXT INFO] Is PR: $isPullRequest, is tag: $isTagBuild, branch name: '$branchName'" -ForegroundColor Green
+
+    if($isTagBuild)
+    {
+      $tagName = $Env:APPVEYOR_REPO_TAG_NAME
+      $isSemVerTag = $tagName -match '^v.*'
+      # We consider version as alpha if it contains the "-alpha" pre-release tag.
+      $isAlphaVerTag = $tagName -match '^v[\d\.]+-alpha[\d\.]*$'
+
+      if(-not $isSemVerTag)
+      {
+        Write-Host "[Tag] Trigger build only - this is not a SemVer tag. Tag name: $tagName" -ForegroundColor Green
+        & .\build.cmd CompleteBuild BuildVersion=git BuildNumber=$($Env:APPVEYOR_BUILD_NUMBER)
+      }
+      # Alpha version is published to the private feed only as tag's purpose is to start the vNext versions count.
+      elseif($isAlphaVerTag)
+      {
+        Write-Host "[Tag] Trigger build & push to private feed. Tag name: $tagName" -ForegroundColor Green
+        & .\build.cmd PublishNuGetPrivate NuGetPrivateKey="$($Env:MYGET_API_KEY)" BuildVersion=git BuildNumber=$($Env:APPVEYOR_BUILD_NUMBER)
+      }
+      else
+      {
+        Write-Host "[Tag] Trigger build & push to public feed. Tag name: $tagName" -ForegroundColor Green
+        & .\build.cmd PublishNuGetPublic NuGetPublicKey="$($Env:NUGET_API_KEY)" BuildVersion=git BuildNumber=$($Env:APPVEYOR_BUILD_NUMBER)
+      }
+    }
+    # Should be checked _before_ branch name, because branch name is set to "base" branch for Pull Requests
+    elseif($isPullRequest -eq $true)
+    {
+      Write-Host "[PR] Trigger verification build only" -ForegroundColor Green
+      & .\build.cmd CompleteBuild BuildVersion=git BuildNumber=$($Env:APPVEYOR_BUILD_NUMBER)
+    }
+    # Push to private feed on commit - so package could be accessed before it's tagged.
+    #elseif($branchName -eq "master") 
```

Comment:
I left this place intentionally, as this is the place where you will enable private builds for the vNext branch.
The reason I kept it is the following:
- it's much easier to enable early builds for `vNext` - just uncomment out a few lines, instead of writing the boilerplate code again and again;
- logic should be located in this particular place and it will lead to mistakes if we place it in another place. If we keep it here, there are less chances that you will make mistake in future;
- this boilerplate code already contains all the necessary parameters you should pass to the FAKE to run trigger the task. I don't remove it, later you will need to create these lines again.

Given that I'm going to consume these lines in `v4` branch after PR is merged (if it even will be...), I believe it's OK to keep them.